### PR TITLE
🕐 feat: Configurable Retention Period for Temporary Chats

### PR DESCRIPTION
**Pull request by ConstantTime**
## Summary

This PR implements configurable retention periods for temporary chats, replacing the hardcoded 30-day limit with flexible configuration options via environment variables or `librechat.yaml`. Users can now set retention periods from 1 day to 1 year based on their needs.

Closes #7673

## Change Type

- [x] New feature (non-breaking change which adds functionality)

### **Core Implementation**
- **New utility**: `api/server/utils/tempChatRetention.js`
  - `getTempChatRetentionDays()`: Retrieves configured retention period with validation
  - `createTempChatExpirationDate()`: Creates expiration dates using configured period
  - Built-in validation (min: 1 day, max: 365 days) with fallback to defaults
- **Updated models**: `api/models/Message.js` and `api/models/Conversation.js` now use configurable retention
- **Schema updates**: Added `temporaryChatRetentionDays` to interface schema in `packages/data-provider/src/config.ts`
- **Configuration integration**: Updated `AppService.js` to make config available to models

### **Configuration Options**
1. **Environment Variable**: `TEMP_CHAT_RETENTION_DAYS=7`
2. **LibreChat.yaml**: `interface.temporaryChatRetentionDays: 14`
3. **Default**: 30 days (maintains backward compatibility)

### **Validation & Safety**
- **Minimum**: 1 day (prevents immediate deletion)
- **Maximum**: 365 days (prevents indefinite retention)
- **Priority**: Config file > Environment variable > Default
- **Error handling**: Invalid values fall back to defaults with warnings

### **Automated Testing**
- **Test suite**: `api/server/utils/__tests__/tempChatRetention.test.js`
- **Coverage**: 12 test cases covering all scenarios
- **Edge cases**: Invalid values, boundary conditions, precedence rules
- **Results**: ✅ All tests passing

#### **Test 1: Default Behavior (30 days)**
```bash
# Configuration: No custom settings
# Expected: 30-day retention
```

**Database Verification:**
```javascript
// Conversation ID: 91e6a871-de82-4240-836e-7f83ea1ad4cc
Created: 2025-06-15T14:59:42.764Z
Expires: 2025-07-15T14:59:45.705Z
Retention: 30.00 days ✅
```

**Results:**
- ✅ Conversation and messages both expire in exactly 30 days
- ✅ Default behavior maintained for backward compatibility

#### **Test 2: Custom 1-Day Retention**
```yaml
# Configuration: librechat.yaml
interface:
  temporaryChatRetentionDays: 1
```

**Database Verification:**
```javascript
// Conversation ID: 4b5629af-fdc2-4a4c-83c2-4619f7e60ac7
Created: 2025-06-15T15:01:28.710Z
Expires: 2025-06-16T15:01:31.451Z
Retention: 1.00 days ✅
```

**Results:**
- ✅ Configuration change immediately applied to new conversations
- ✅ Precise 24-hour retention period
- ✅ Both conversation and messages have consistent expiration dates
- ✅ MongoDB TTL index will auto-delete expired records

#### **Test 3: Configuration Priority**
```bash
# Environment: TEMP_CHAT_RETENTION_DAYS=15
# Config file: temporaryChatRetentionDays: 1
# Expected: Config file wins (1 day)
```

**Results:**
- ✅ Config file setting (1 day) took precedence over environment variable
- ✅ Priority system working correctly

### **Database Schema Verification**
```javascript
// Conversation Document
{
  conversationId: "4b5629af-fdc2-4a4c-83c2-4619f7e60ac7",
  expiredAt: ISODate('2025-06-16T15:01:31.451Z'), // ✅ 1-day retention
  createdAt: ISODate('2025-06-15T15:01:28.710Z'),
  // ... other fields
}

// Message Documents
{
  messageId: "783e67d9-a6d4-403d-a91f-03eb6b7ad090",
  expiredAt: ISODate('2025-06-16T15:01:30.795Z'), // ✅ 1-day retention
  text: "Testing temp chat",
  // ... other fields
}
```

## Documentation

- **Updated**: `librechat.example.yaml` with configuration example and comments
- **Added**: Comprehensive JSDoc comments in utility functions
- **Included**: Validation ranges and error handling documentation

## Backward Compatibility

- ✅ **Fully backward compatible**: Existing installations continue working unchanged
- ✅ **Default behavior preserved**: 30-day retention when no configuration is provided
- ✅ **No breaking changes**: All existing temporary chats maintain their expiration dates

## Performance Impact

- ✅ **Minimal overhead**: Configuration loaded once at startup
- ✅ **Efficient validation**: Simple numeric range checks
- ✅ **Database optimized**: Uses existing MongoDB TTL indexes for cleanup

## Security Considerations

- ✅ **Input validation**: Prevents invalid retention periods
- ✅ **Bounds enforcement**: Prevents both immediate deletion and indefinite retention
- ✅ **Error handling**: Graceful fallback to safe defaults

### CODE REVIEW

**Comment by ConstantTime** [comment]
Addressed the above issue and build is passing now.

File: packages/data-provider/src/config.ts [hunk 1]
```diff
@@ -503,6 +503,7 @@ export const intefaceSchema = z
     prompts: z.boolean().optional(),
     agents: z.boolean().optional(),
     temporaryChat: z.boolean().optional(),
+    temporaryChatRetentionDays: z.number().min(1).max(365).optional(),
```
Comment (danny-avila) [inline packages/data-provider/src/config.ts]: documentation update needed: availableTools

Comment (danny-avila) [reply packages/data-provider/src/config.ts]: also, nitpick: rename to `temporaryChatRetention` and make unit to hours for more granular customization

**Review by danny-avila — CHANGES_REQUESTED** [review]

File: packages/data-provider/src/config.ts (hunk 1 above)
Comment (ConstantTime) [reply packages/data-provider/src/config.ts]: Name update is done and documentation added [here](https://github.com/LibreChat-AI/librechat.ai/pull/334).

**Comment by alifa98** [comment]
thank you @ConstantTime for adding this configuration. 

I have a question and it is a little bit unrelated directly to this pull request, but I am curious if there is any mechanism in the code base to use this field of expiry date? or is it purely delegated to users to implement their own clean up system? 

we can continue chatting in the following discussion topic if you think that we shouldn't discuss it here as it's an indirect problem related to this PR:

https://github.com/danny-avila/LibreChat/discussions/7674

**Comment by ConstantTime** [comment]
> thank you @ConstantTime for adding this configuration.
> 
> I have a question and it is a little bit unrelated directly to this pull request, but I am curious if there is any mechanism in the code base to use this field of expiry date? or is it purely delegated to users to implement their own clean up system?
> 
> we can continue chatting in the following discussion topic if you think that we shouldn't discuss it here as it's an indirect problem related to this PR:
> 
> #7674

Hey @alifa98 
No it's fine, we can discuss this here. Users don't have to worry about deletion mechanisms. Seems like this is being taken care by mongoDB itself. Example: https://github.com/danny-avila/LibreChat/blob/main/packages/data-schemas/src/schema/convo.ts#L46-L48

The `expiredAt`/`expiresAt` fields are actively used by MongoDB's built-in TTL feature to automatically delete expired documents without any manual intervention required. We just had to make that parameter configurable here.

**Comment by danny-avila** [comment]
thanks @ConstantTime !

File: api/server/utils/tempChatRetention.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+ * Maximum allowed retention period in hours (1 year = 8760 hours)
+ */
+const MAX_RETENTION_HOURS = 8760;
+
+/**
+ * Gets the temporary chat retention period from environment variables or config
+ * @param {TCustomConfig} [config] - The custom configuration object
+ * @returns {number} The retention period in hours
+ */
+function getTempChatRetentionHours(config) {
```
Comment (danny-avila) [inline api/server/utils/tempChatRetention.js]: Can you move this to `packages/api` ? All new backend-specific code should live, as much as possible.

Then, it gets imported from `@librechat/api` after its compiled i.e. `npm run build:api`

**Review by danny-avila — CHANGES_REQUESTED** [review]

File: api/server/utils/tempChatRetention.js (hunk 2 above)
Comment (ConstantTime) [reply api/server/utils/tempChatRetention.js]: Done!

**Comment by danny-avila** [comment]
Thanks for your work on this! Needed some additional changes that required moving more things over to `packages/api`:

Moved here including your original commits:
https://github.com/danny-avila/LibreChat/pull/8056
